subsys/net/lib/socks/socks.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019 Antmicro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_socks, CONFIG_SOCKS_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <errno.h>
+#include <net/socket.h>
+#include <net/socks.h>
+
+#include "socks_internal.h"
+
+static int socks5_tcp_send(int fd, u8_t *data, u32_t len)
+{
+	u32_t offset = 0U;
+	int ret;
+
+	while (offset < len) {
+		ret = send(fd, data + offset, len - offset, 0);
+		if (ret < 0) {
+			return -errno;
```

Comment:
Why return errno here? Just return ret, caller can look up actual error in errno itself. That's standard POSIX protocol.